src/events/index.test.ts

```diff
@@ -51,20 +52,20 @@ describe('allEvents', () => {
 		// Be sure to clear all the auto-added event handlers first, or else they'll mess up our count.
 		// Casting a read-only list into a regular list is bad practice, but this is for testing purposes.
 		// Don't do this at home.
-		(allEventHandlers as Map<keyof ClientEvents, EventHandler>).clear();
+		(allEventHandlers as Map<string, unknown>).clear();
 
-		const fakeReadyEvent: EventHandler = {
+		const fakeReadyEvent: EventHandler<'ready'> = {
 			name: 'ready',
```

Comment:
Is this line unnecessary when EventHandler defines `name = T`? (Same for `messageCreate` below)

Reply:
You'd think so, but TypeScript throws fits when I try:

![Screen Shot 2022-10-05 at 15 32 54](https://user-images.githubusercontent.com/1500092/194168005-95040cc0-c6a3-4c3a-a16f-8e49da8d9336.png)

Reply:
That's why the helper function, to reduce duplicating the `name` value

Reply:
Making `keyof ClientEvents` the default fixes this tho, since we cast these object to `EventHandler<keyof ClientEvents>` in the next statement anyway.